rfcs/text/0030/process.yml

```diff
@@ -0,0 +1,191 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+- name: process
+  reusable:
+    top_level: true
+    expected:
+      - at: process
+        as: entry_leader
+        short_override: Information about the entry session leader process.
+      - at: process
+        as: session_leader
+        short_override: Information about the session leader process.
+      - at: process
+        as: group_leader
+        short_override: Information about the process group leader process. Undefined if same as process.
+
+  fields:
+
+    - name: ppid
```

Comment:
As of ECS 8.0, we have removed `process.ppid` https://github.com/elastic/ecs/pull/1596 in favor of using `process.parent.*`

Reply:
process.parent is not sufficient when you consider the other ancestor process objects like process.session_leader do not have a process.session_leader_parent from which that ppid may be fetched.

Reply:
hey @kgeller thanks for the feedback. Is it ok if I resolve this (ie process.ppid goes back into ECS) and same for ppid_start?

Reply:
I think this will warrant a discussion within the ecs team, but you are welcome to resolve it in the meantime and we can followup 

cc @ebeahan @djptek

Reply:
You can define a reuse like to create a `process.session_leader.parent.pid` field:

```yaml
      - at: process
        as: session_leader
      - at: process.session_leader
        as: parent
```

`process.ppid` preceded the `process.parent.*` field reuse and, like @kgeller mentioned, is being removed in ECS 8.0 in favor of `process.parent.pid` (I realized this detail wasn't evident in the docs and have opened https://github.com/elastic/ecs/issues/1693 to address).   

Having two fields for the same purpose is something ECS tries to avoid when possible. The situation creates user confusion (when to use `process.ppid` over `process.parent.pid`) or creates missed detections (a detection rule is looking at `process.parent.pid` but the user populates `process.ppid`).

Reply:
@ebeahan we're aware ppid is slated to go away, which is why we're suggesting it be re-added along with *_starttime which help disambiguate them when pid-wrap occurs.  As mentioned in the comment above some of these ppid values would not be those of the included processes (parent, session_leader, group_leader) - e.g. session_leader ppid would correspond to a process that is not in the (parent, session_leader, group_leader) - ie we're not including all ancestors of the process.    The ppid values allow rendering a session view with a full process tree (~foreign key to find the parent exec process event).

Reply:
if it's ok to use anemic instances of process objects (with just a few fields included) to act as foreign keys, and this approach is preferred to ppid+parent_starttime,  we can refactor a few things in process.

Reply:
In the ECS docs, this would appear as full reuse, but in the actual implementation, there are ways to use the `--subset` option in the tooling to limit what fields are used (The `endpoint-package` has [good examples](https://github.com/elastic/endpoint-package/tree/master/custom_subsets) of what this filtering looks like).

Reply:
Updated to reflect this discussion. Also, I added a comment about deprecating 'pgid' as it will now be redundant since process.group_leader.pid will be the same value. Which matches the pattern wrt ppid decisions.

Reply:
Small addition on top of the above. process.parent.group_leader nesting is possible now. This was added due to a requirement that we know the pgid of the parent. It allows us to infer if the process was triggered by a real user.